api/src/main/java/org/openmrs/Obs.java

```diff
@@ -1111,7 +1111,22 @@ public void setValueAsString(String s) throws ParseException {
 			} else if ("ST".equals(abbrev)) {
 				setValueText(s);
 			} else {
-				throw new RuntimeException("Don't know how to handle " + abbrev);
+				String[] supportedFormats = { "yyyy-MM-dd'T'HH:mm:ss.SSSZ", "yyyy-MM-dd'T'HH:mm:ss.SSS",
+					"yyyy-MM-dd'T'HH:mm:ssZ", "yyyy-MM-dd'T'HH:mm:ssXXX", "yyyy-MM-dd'T'HH:mm:ss", "yyyy-MM-dd HH:mm:ss"};
+				boolean flag = false;
+				for (int i = 0; i < supportedFormats.length; i++) {
+					try {
+						DateFormat dateFormat = new SimpleDateFormat(supportedFormats[i]);
+						setValueDatetime(dateFormat.parse(s));
+						flag = false;
+						break;
+					}
+					catch (Exception ex) {
```

Comment:
Isn't it better to specify the exception instead of catching any exception?

Reply:
Given that Exception is a super class, I don't think it should cause much of an issue?